src/NuGet.Clients/NuGet.VisualStudio.Common/Services/NuGetFeatureFlagService.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(INuGetFeatureFlagService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class NuGetFeatureFlagService : INuGetFeatureFlagService
+    {
+        private readonly IEnvironmentVariableReader _environmentVariableReader;
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
+        private readonly Microsoft.VisualStudio.Threading.AsyncLazy<IVsFeatureFlags> _ivsFeatureFlags;
+        private readonly Dictionary<string, bool> _featureFlagCache;
+
+        [ImportingConstructor]
+        public NuGetFeatureFlagService()
+            : this(EnvironmentVariableWrapper.Instance, AsyncServiceProvider.GlobalProvider)
+        {
+            // ensure uniqueness.
+        }
+
+        internal NuGetFeatureFlagService(IEnvironmentVariableReader environmentVariableReader, IAsyncServiceProvider asyncServiceProvider)
+        {
+            _environmentVariableReader = environmentVariableReader ?? throw new ArgumentNullException(nameof(environmentVariableReader));
+            _asyncServiceProvider = asyncServiceProvider ?? throw new ArgumentNullException(nameof(asyncServiceProvider));
+            _ivsFeatureFlags = new(() => _asyncServiceProvider.GetServiceAsync<SVsFeatureFlags, IVsFeatureFlags>(), NuGetUIThreadHelper.JoinableTaskFactory);
+            _featureFlagCache = new();
+        }
+
+        public async Task<bool> IsFeatureEnabledAsync(NuGetFeatureFlagConstants featureFlag)
+        {
+            GetEnvironmentVariablesForFeature(featureFlag, out bool isFeatureForcedEnabled, out bool isFeatureForcedDisabled);
```

Comment:
Out of curiosity, what is the difference between `isFeatureForcedEnabled` and `isFeatureForcedDisabled`? I assume a feature can be turned off by the user if it enabled by default. My understanding is that one variable would be sufficient to check if the feature is turned on or off. It could be possible that, I am missing something here.